src/zenml/steps/utils.py

```diff
@@ -38,57 +131,183 @@ def resolve_type_annotation(obj: Any) -> Any:
         obj: The object to resolve.
 
     Returns:
-        The non-generic class for generic aliases of the typing module.
+        The non-generic class for generic aliases of the typing module and
+        optional annotation metadata if it exists.
     """
     origin = pydantic_typing.get_origin(obj) or obj
 
-    if pydantic_typing.is_union(origin):
-        return obj
+    if origin is Annotated:
+        annotation, *_ = typing.get_args(obj)
+        output_name = validate_annotation_metadata(obj)
 
-    return origin
+        resolved_annotation, _ = resolve_type_annotation(annotation)
+        return resolved_annotation, output_name
 
+    elif pydantic_typing.is_union(origin):
+        return obj, None
 
-def get_args(obj: Any) -> Tuple[Any, ...]:
-    """Get arguments of a Union type annotation.
+    return origin, None
+
+
+def validate_annotation_metadata(annotation: Any) -> str:
+    """Validates annotation metadata.
+
+    Args:
+        annotation: The type annotation, must be of type `Annotated[...]`
```

Comment:
This first part of the docstring could be clearer on what the function does

Reply:
Maybe also rename to `get_output_name_from_annotation_metadata()`?

Reply:
Renamed it